### PR TITLE
Fixed error when filing didn't have a context

### PR DESCRIPTION
**Pull request by sagesmith-wf**
#### Changes
- Check that facts.context is not None because it was throwing an error
- Defensive check that fact is not None
- Defensive check that fact.context.segDimValue is not None
- Defensive check that fact.context.segDimValue.values() is not None

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

File: tests/unit_tests/util/test_facts.py [hunk 1]
```diff
@@ -311,4 +620,35 @@ def test_member_qnames(self):
...
+        """
+        Tests to make sure that a Fact with a None type context is not valid
+        """
+        fact = Mock(decimals='-1', value='670', xValue=670, precision=None)
+        fact.context = None
+        self.assertFalse(fact_lib.fact_components_valid(fact))
+
+    def test_fact_components_valid_on_none_type_segDimValue(self):
+        """
+        Tests to make sure that a Fact.context with a None type context is not valid
```
Comment (jasonleinbach-wf) [inline tests/unit_tests/util/test_facts.py]: Think this should be "Tests to make sure that a Fact.context with a None type segDimValues is not valid"

**Comment by derekgengenbacher-wf** [comment]
+1

File: tests/unit_tests/util/test_facts.py [hunk 2]
```diff
@@ -311,4 +620,35 @@ def test_member_qnames(self):
...
+        """
+        Tests to make sure that a Fact with a None type context is not valid
+        """
+        fact = Mock(decimals='-1', value='670', xValue=670, precision=None)
+        fact.context = None
+        self.assertFalse(fact_lib.fact_components_valid(fact))
+
+    def test_fact_components_valid_on_none_type_segDimValue(self):
+        """
+        Tests to make sure that a Fact.context with a None type sedDimValues is not valid
```
Comment (jasonleinbach-wf) [inline tests/unit_tests/util/test_facts.py]: Typo in the update!  'segDimValues'.  :-)

**Comment by jasonleinbach-wf** [comment]
+1

**Comment by derekgengenbacher-wf** [comment]
+1 refresh

**Comment by jasonleinbach-wf** [comment]
+10 - removed a context attribute from a DPED fact and verified no error on validation using the DQC plugin at this branch.

**Pull request merged**